src/main/kotlin/me/partlysanestudios/partlysaneskies/features/skills/PetAlert.kt

```diff
@@ -0,0 +1,232 @@
+            return
+        }
+
+        if (onCooldown(lastMuteTime, (config.petAlertMuteTime * 60L * 1000L).toLong())) {
+            return
+        }
+
+        if (!onCooldown(lastSoundTime, 750)) {
+            minecraft.soundHandler.playSound(PositionedSoundRecord.create(ResourceLocation("partlysaneskies", "bell")))
+
+            if (config.incorrectPetForMinionAlertSiren) {
+                minecraft.soundHandler.playSound(PositionedSoundRecord.create(ResourceLocation("partlysaneskies", "airraidsiren")))
+            }
+
+            lastSoundTime = time
+        }
+
+        if (!onCooldown(lastMessageSendTime, 3000)) {
+            val message: IChatComponent = ChatComponentText("${PartlySaneSkies.CHAT_PREFIX}§cYOU CURRENTLY HAVE $petName§c SELECTED AS YOUR PET. YOU WANTED TO UPGRADE $selectedPetName.\n§dClick this message or run /mutepetalert to mute the alert for ${config.petAlertMuteTime} minutes.")
+            message.chatStyle.setChatClickEvent(ClickEvent(ClickEvent.Action.RUN_COMMAND, "/mutepetalert"))
+            minecraft.ingameGUI.chatGUI.printChatMessage(message)
+            lastMessageSendTime = time
+        }
+    }
+
+    private fun isMinionGui(): Boolean {
+        if (minecraft.currentScreen !is GuiChest) {
+            return false
+        }
+
+        val inventories = minecraft.currentScreen.getSeparateUpperLowerInventories()
+
+        if (inventories[0] == null) {
+            return false
+        }
+
+        val upper = inventories[0] ?: return false
+        val inventoryNameMatches = upper.displayName.formattedText.removeColorCodes().contains("Minion")
+        if (!inventoryNameMatches) {
+            return false
+        }
+
+//        Gets the slot where the minion head is supposed to be
+        val minionHeadSlot = upper.getStackInSlot(4) ?: return false
+        val displayName = minionHeadSlot.displayName.removeColorCodes()
+        return displayName.contains("Minion")
+    }
+
+
+    fun favouritePet() {
+        if (!isSkyblock()) {
+            return
+        }
+
+
+
+        if (!isPetGui()) {
+            return
+        }
+
+        if (minecraft.currentScreen is AuctionHouseGui) {
+            return
+        }
+
+        val container = minecraft.currentScreen as GuiContainer
+        val slot = container.slotUnderMouse ?: return
+        val item = slot.stack
+
+        if (item.getItemId().isEmpty()) {
+            return
+        }
+
+        val petName = parsePetNameFromItem(item.displayName)
+        config.selectedPet = petName
+        sendClientMessage("Set $petName as your favorite pet.")
+        config.save()
+    }
+
+    fun registerCommand() {
+        PSSCommand("mutepetalert")
+            .setDescription("Mutes the pet alert for a set amount of minutes")
+            .setRunnable { _: ICommandSender, _: Array<String> ->
+                sendClientMessage("§bPet alert has been muted for " + config.petAlertMuteTime + " minutes.")
```

Comment:
Add a pluralize(int, string) method, that adds a "s" at the end of the word if the number is not 1